actionview/lib/action_view/helpers/asset_url_helper.rb

```diff
@@ -168,7 +168,7 @@ def asset_path(source, options = {})
       #   asset_url "application.js", host: "http://cdn.example.com" # => http://cdn.example.com/assets/application.js
       #
       def asset_url(source, options = {})
-        path_to_asset(source, options.merge(:protocol => :request))
+        path_to_asset(source, { protocol: default_asset_protocol }.merge(options))
```

Comment:
I'm not sure that we should say `:request` here even if there's no `request` in scope - in fact I think that will break. I preserved that existing behavior (perhaps it's invalid to call `asset_url` in that case anyway?), but if it's ok to fall back to `:relative` when `request` is undefined, as we do in the `compute_asset_host` case, we could simplify the API to the `default_asset_protocol` function.